### PR TITLE
Hexmap

### PR DESCRIPTION
**Pull request by bonndan**
provides a browser-based navigable svg map with improved rendering of relations/paths
adresses #62

### CODE REVIEW

File: src/main/java/de/bonndan/nivio/output/jgraphx/FinalGraph.java [hunk 1]
```diff
@@ -9,31 +9,47 @@
...
 import java.util.stream.Stream;
 
 import static de.bonndan.nivio.model.Status.UNKNOWN;
 import static de.bonndan.nivio.output.Color.getGroupColor;
 
-public class FinalGraph {
+/**
+ * This class is responsible for rendering services and groups nicely with bells and whistles.
+ * <p>
+ * It receives
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/output/jgraphx/FinalGraph.java]: "It receives..."

File: src/main/app/src/NLabel.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+                <foreignObject width={width} height="220" y={padding} x={size + padding } style={style}>
+                    <div className="details">
+                        {item.landscapeItem.description && (<div >"{item.landscapeItem.description}"<br /></div>)}
+                        {item.landscapeItem.owner && (<div >Owner: {item.landscapeItem.owner}</div>)}
+                        {item.landscapeItem.team && (<div >Team: {item.landscapeItem.team}</div>)}
+                        {item.landscapeItem.contact && (<div >Contact: {item.landscapeItem.contact}</div>)}
+                        {item.landscapeItem.software && (<div >Software: {item.landscapeItem.software}</div>)}
+                        {item.landscapeItem.version && (<div >Version: {item.landscapeItem.version}</div>)}
+                        {item.landscapeItem.lifecycle && (<div >Lifecycle: {item.landscapeItem.lifecycle}</div>)}
```
Comment (basseur) [inline src/main/app/src/NLabel.js]: Remove spaces within `<div>`.

File: src/main/app/src/Nexagon.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {Hex, HexUtils} from 'react-hexgrid';
+
+
+class Nexagon extends Component {
```
Comment (basseur) [inline src/main/app/src/Nexagon.js]: Is this correct or was it supposed to be named `NHexagon`?

File: src/main/app/src/PathFinder.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            if (possibleSteps.length === 0) { //TODO wrong, wont go back
+                path.close()
+            } else {
+
+                //prolong path and create clones if there are more possibilties
+                let template = new TilePath();
+                template.tiles = path.tiles;
+                path.tiles.push(possibleSteps.shift());
+
+                for (var i = 0; i < possibleSteps.length; i++) {
```
Comment (basseur) [inline src/main/app/src/PathFinder.js]: `let` instead of `var`

File: src/main/app/src/TilePath.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+        return false;
+    }
+
+    getSpeed() {
+        return this.tiles.length - this.bends.length;
+    }
+}
+
+export default TilePath;
```
Comment (basseur) [inline src/main/app/src/TilePath.js]: You should really run _Reformat Code_ on this file (`Ctrl`+`Alt`+`L`). 🙃 Loads of missing spaces.

File: src/main/java/de/bonndan/nivio/output/jgraphx/FinalGraph.java [hunk 6]
```diff
@@ -9,31 +9,47 @@
...
+
 import de.bonndan.nivio.model.*;
 import de.bonndan.nivio.output.*;
+import de.bonndan.nivio.output.Rendered;
 import de.bonndan.nivio.util.RootPath;
+
+import de.bonndan.nivio.output.Icon;
+import de.bonndan.nivio.output.IconService;
+import de.bonndan.nivio.output.LocalServer;
+
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/output/jgraphx/FinalGraph.java]: Why do you import `Rendered`, `Icon`, `IconService` and `LocalServer `specifically if you already import `de.bonndan.nivio.output.*`?

File: src/main/java/de/bonndan/nivio/output/jgraphx/FinalGraph.java [hunk 7]
```diff
@@ -9,31 +9,47 @@
...
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.util.StringUtils;
 
 import java.io.File;
 import java.net.MalformedURLException;
 import java.util.*;
+import java.util.List;
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/output/jgraphx/FinalGraph.java]: ...again.

File: src/main/java/de/bonndan/nivio/output/jgraphx/RenderedXYMapFactory.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+            long x;
+            long y;
+            if (cell.getParent().getGeometry() != null) {
+                x = Math.round(geometry.getX() + cell.getParent().getGeometry().getX());
+                y = Math.round(geometry.getY() + cell.getParent().getGeometry().getY());
+            } else {
+                x = Math.round(geometry.getX());
+                y = Math.round(geometry.getY());
+            }
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/output/jgraphx/RenderedXYMapFactory.java]: Can't you extract this into a method (using some kind of Point)? It get's repeated in lines 93-102.

File: src/main/java/de/bonndan/nivio/output/map/Hex.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    static public ArrayList<Hex> diagonals = new ArrayList<Hex>() {{
+        add(new Hex(2, -1, -1));
+        add(new Hex(1, -2, 1));
+        add(new Hex(-1, -1, 2));
+        add(new Hex(-2, 1, 1));
+        add(new Hex(-1, 2, -1));
+        add(new Hex(1, 1, -2));
+    }};
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/output/map/Hex.java]: Move static fields to the top of the class.

File: src/main/java/de/bonndan/nivio/output/map/MapItem.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
+package de.bonndan.nivio.output.map;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import de.bonndan.nivio.model.LandscapeItem;
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/output/map/MapItem.java]: remove

**Comment by bonndan** [comment]
Great review, thanks

File: src/main/java/de/bonndan/nivio/output/jgraphx/FinalGraph.java (hunk 1 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/jgraphx/FinalGraph.java]: done

File: src/main/java/de/bonndan/nivio/output/map/Hex.java (hunk 9 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/map/Hex.java]: done

File: src/main/java/de/bonndan/nivio/output/jgraphx/RenderedXYMapFactory.java (hunk 8 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/jgraphx/RenderedXYMapFactory.java]: done.

File: src/main/app/src/NGroup.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+        let startPoint = HexUtils.hexToPixel(start, layout);
+        let endPoint = HexUtils.hexToPixel(end, layout);
+        let width = endPoint.x - startPoint.x + 4 * size;
+        let height = endPoint.y - startPoint.y + 5*size;
+        return (
+            <g>
+                <rect x={startPoint.x - 2 * size} y={startPoint.y - 2* size} rx="50" ry="50" width={width} height={height}
+                      style={cellStyle}
+                      stroke={fillId} fill={fillId} className="group"/>
+                <text x={startPoint.x - 2 * size + width/2} y={startPoint.y + height - 3*size} fill={fillId}  font-size="24" width={width} textAnchor={'middle'}>{group.name}</text>
```
[the remark this answers is not in the document]
Comment (bonndan) [reply src/main/app/src/NGroup.js]: can be refactored in develop branch

File: src/main/app/src/Nexagon.js (hunk 3 above)
Comment (bonndan) [reply src/main/app/src/Nexagon.js]: The whole naming can be refactored later

File: src/main/app/src/PathFinder.js (hunk 4 above)
Comment (bonndan) [reply src/main/app/src/PathFinder.js]: done

**Pull request merged**